hathor/event/websocket/request.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from pydantic import NonNegativeInt
+
+from hathor.utils.pydantic import BaseModel
+
+
+class StreamRequest(BaseModel):
+    """Class that represents a client request to stream events.
+
+    Args:
+        last_received_event_id: The ID of the last event successfully processed by the requesting client.
+        window_size_increment: The amount of events the client is able to process next.
+    """
+    last_received_event_id: Optional[NonNegativeInt]
```

Comment:
Why is it optional? I feel the client can just repeat the last value if nothing has changed.

Reply:
> Why is it optional?

Because the first time the client connects to the server, it hasn't received any event, and that's represented by `None` (also when the client wants to re-receive events from the beginning).

I believe that's simpler than creating two different request schemas (one for starting the stream and another one for ACK), especially considering that the "stop streaming" message can also use the same schema (setting `window_size = 0`). What do you think?

> I feel the client can just repeat the last value if nothing has changed.

If the client sends two messages with the same `last_received_event_id `, that should make the server resend all events from that ID, right?

Reply:
It seems we have a misunderstanding of how it should work. As the design was not updated, I feel we should have a quick meeting to align about it tomorrow. Anyway, I'll add more details about my view on another thread.

Reply:
We covered the expected behavior on a meeting, and it has already been developed. Good job, @glevco . Thanks!